render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java

```diff
@@ -137,6 +148,61 @@ private void renderJPEG(final ImageSegment imageSegment, final Graphics2D target
         });
     }
 
+    private void renderJPEG2k(final ImageSegment imageSegment, final Graphics2D targetGraphic) throws IOException {
+        final ImageReader reader = getImageReader("image/jp2");
+        reader.setInput(imageSegment.getData(), true, true);
+        final ImageReadParam param = reader.getDefaultReadParam();
+
+        if (ImageRepresentation.MULTIBAND.equals(imageSegment.getImageRepresentation())) {
+            final int[] sourceBands = getSourceBands(imageSegment);
+            param.setSourceBands(sourceBands);
+        }
+
+        processBlocks(imageSegment, (r, c) -> {
+                    Rectangle rect = new Rectangle((int) (c * imageSegment.getNumberOfPixelsPerBlockVertical()),
+                            (int) (r * imageSegment.getNumberOfPixelsPerBlockHorizontal()),
```

Comment:
I pulled this code out into a test class to run the scenario you outlined.  The x and y calculations had PPBh and PPBv swapped.  Changed this.